### PR TITLE
add geographic aggregates to grad rate and grad count data

### PR DESCRIPTION
**Pull request by chiouey**
## proposed changes
- create `ward_grate_aggs` and `ward_gcount_aggs` in the style of `ward_enr_aggs`
- write return type and ground truth tests

## test
`testthat::test_file("tests/testthat/test_geo.R")`

### CODE REVIEW

File: R/geo.R [hunk 1]
```diff
@@ -167,4 +167,178 @@ ward_enr_aggs <- function(df) {
...
+ward_parcc_aggs <- function(list_of_dfs) {
+   
+   df <- list_of_dfs %>%
+      bind_rows() %>% # convert to df
+      enrich_school_latlong() %>%
+      enrich_school_city_ward()
+   
+   df <- df %>%
+      filter(!is.na(ward) #,
+             #subgroup == "special_education",
```
Comment (almartin82) [inline R/geo.R:189]: probably had these in here for testing / dev purposes?  would be good to clean these up before merging the final version

File: R/geo.R [hunk 2]
```diff
@@ -167,4 +167,178 @@ ward_enr_aggs <- function(df) {
...
+      group_by(
+         testing_year,
+         assess_name, test_name,
+         county_id, county_name,
+         district_id, district_name,
+         ward,
+         grade,
+         subgroup, subgroup_type
+      ) %>%
+      summarize( # this should be replaced with parcc_aggregate_calcs()
```
Comment (almartin82) [inline R/geo.R:204]: did we try doing this with parcc_aggregate_calcs?

File: R/grate.R [hunk 3]
```diff
@@ -114,7 +114,12 @@ process_grate <- function(df, end_year) {
   if ('program_name' %in% names(df)) {
     df$program_name <- ifelse(df$program_name %in% c('Total', 'TOTAL'), 'Total', df$program_name) 
   }
-  df$school_id <- ifelse(df$school_id == "999.000000", "999", df$school_id)
+  
+  df <- df %>%
+     mutate(school_id = case_when(school_id == "999.000000" ~ "999",
+                                  school_id == "888" & end_year == 2019 ~ "999",
```
Comment (almartin82) [inline R/grate.R:120]: oh my!  they added another dummy code for district level aggregate?!

File: R/grate.R [hunk 4]
```diff
@@ -373,13 +385,20 @@ grad_file_group_cleanup <- function(group) {
...
     group %in% c('american indian or alaska native', 'american_indian') ~ 'american indian',
     group %in% c('black or african american') ~ 'black',
-    group %in% c('economically_disadvantaged') ~ 'economically disadvantaged',
+    group %in% c('economically_disadvantaged',
+                 'economically disadvantaged students') ~ 'economically disadvantaged',
     group %in% c('english learners', 'limited_english_proficiency') ~ 'limited english proficiency',
-    group %in% c('two or more race', 'two_or_more_races') ~ 'multiracial',
+    group %in% c('two or more race', 'two_or_more_races', 'two or more races') ~ 'multiracial',
     group %in% c('native hawaiian or pacific islander', 'pacific_islander', 'native_hawaiian') ~ 'pacific islander',
+    group %in% c('asian, native hawaiian, or pacific islander') ~ 'asian', # 2019 groups a and pi together?!
```
Comment (almartin82) [inline R/grate.R:393]: 🙃

File: R/geo.R (hunk 1 above)
Comment (chiouey) [reply R/geo.R]: oops -- fixed adb4377396b89189e93b10ae405f31bc36ba9d0f

File: R/geo.R (hunk 2 above)
Comment (chiouey) [reply R/geo.R]: successfully tried (and tested!) fc950dd353657b9403f0670fef4241de0e5ce251

**Comment by almartin82** [comment]
fixes look good, and tests passing 👍

**Pull request merged**